isis/src/base/apps/spkwriter/SpkKernelWriter.h

```diff
@@ -122,7 +122,11 @@ class SpkKernelWriter : public KernelWriter<SpkKernel> {
     }
 
     void  k_close(SpiceInt &handle) {
-      if ( handle > 0 ) { spkcls_c(handle); }
+      if ( handle != 0 ) {
+        NaifStatus::CheckErrors();
```

Comment:
What does this do?

Reply:
the Naif C-API (the API we use to do pretty much anything with SPICE kernels provided by NAIF) has a wonky C-style error checking system where if something goes wrong you need to make an explicit call to an API function that sets or gets the error. 

You can read more if you low key hate yourself and your free time: https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/req/error.html